presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/JdbcPrestoAction.java

```diff
@@ -131,14 +128,13 @@ private <T> T execute(Statement statement, QueryStage queryStage, StatementExecu
     private <T> T executeOnce(Statement statement, QueryStage queryStage, StatementExecutor<T> statementExecutor)
     {
         String query = formatSql(statement, Optional.empty());
-        String clientInfo = new ClientInfo(
-                testId,
-                testName,
-                verificationContext.getSourceQueryName(),
-                verificationContext.getSuite(),
-                queryStage.name()).serialize();
-
-        try (PrestoConnection connection = getConnection(queryStage, clientInfo)) {
+        String clientInfoJsonString = clientInfo
+                .setSourceQueryName(verificationContext.getSourceQueryName())
```

Comment:
I'm not sure whether this is thread safe. From what I understand the `ClientInfo` instance is a singleton instance being injected.

Reply:
Checked in debug mode, each thread will have a new clientinfo instance. 
From code, each query will get a new instance of PrestoAction. So i think it is thread safe here.